views-core/src/main/java/io/micronaut/views/http/RawModelAndViewMessageBodyHandler.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.http;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.io.buffer.ByteBuffer;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.type.Headers;
+import io.micronaut.core.type.MutableHeaders;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.body.RawMessageBodyHandler;
+import io.micronaut.http.codec.CodecException;
+import io.micronaut.http.context.ServerRequestContext;
+import io.micronaut.views.ModelAndView;
+import io.micronaut.views.ModelAndViewRenderer;
+import io.micronaut.views.exceptions.ViewRenderingException;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * A message body handler for {@link ModelAndView}.
+ *
+ * @param <T> The model type
+ * @author Tim Yates
+ * @since 6.0.0
+ */
+@Singleton
+@Requires(classes = HttpRequest.class)
+public class RawModelAndViewMessageBodyHandler<T> implements RawMessageBodyHandler<ModelAndView<T>> {
```

Comment:
annotate with @Internal
```suggestion
final class RawModelAndViewMessageBodyHandler<T> implements RawMessageBodyHandler<ModelAndView<T>> {
```

Reply:
I think we can remove the Requires(classes = HttpRequest.class) since I added a `package-info.java` with `@Configuration` to the `io.micronaut.views.http` package